tests/unit/base/MockObject.h

```diff
@@ -0,0 +1,27 @@
+#include <shogun/base/SGObject.h>
+#include <shogun/lib/SGVector.h>
+
+namespace shogun
+{
+class CMockObject : public CSGObject
+{
+public:
+    CMockObject() : CSGObject()
+    {
+        init_params();
+    }
+
+    const char* get_name() const { return "MockObject"; }
+
+protected:
+    void init_params()
+    {
+        int32_t integer = 0;
+        float64_t decimal = 0.0;
+        add("vector", SGVector<float64_t>());
```

Comment:
this add could be called different.
register maybe? It should be obvious that this registers a parameter within the object (and it cannot be called from outside)